config.py

```diff
@@ -0,0 +1,301 @@
+                blacklist(dirpath)
+                ## Uncomment this, if you want to remove CATKIN_IGNOREs in the packages when blacklisting the whole dir. Comment the next line then though.
+                # wl_packages.update(pkgs_in_dir)
+                dirnames[:] = []
+            else:
+                whitelist(dirpath)
+
+        if '.git' in dirnames:
+            # don't look into .git directories
+            del dirnames[dirnames.index('.git')]
+
+def ask_permission():
+    """Ask for permission to create and delete CATKIN_IGNORE files."""
+    if len(bl_pkg_paths) != 0:
+        print("\nThe script will CREATE CATKIN_IGNOREs in these packages:")
+        print_pkg_list(bl_pkg_paths)
+
+    if len(wl_pkg_paths) != 0:
+        print("\nThe script will DELETE CATKIN_IGNOREs in these packages:")
+        print_pkg_list(wl_pkg_paths)
+
+    answer = ""
+    possibilities = ['Y', 'y', 'N', 'n']
+    while answer not in possibilities:
+        if answer != "":
+            print("Please type one of '{}', '{}', '{}' or '{}'".format(*possibilities))
+        answer = raw_input("Do you want to proceed? (Y/n)") or "Y"
+    if answer in ['N', 'n']:
+        return False
+    return True
+
+def execute():
+    """Create CATKIN_IGNOREs in the blacklisted paths and delete those in the whitelisted paths."""
+    for dirpath in bl_pkg_paths.keys():
+        open(os.path.join(dirpath, 'CATKIN_IGNORE'), 'w')
+
+    for dirpath in wl_pkg_paths.keys():
+        os.remove(os.path.join(dirpath, 'CATKIN_IGNORE')) 
+
+def profile_str():
+    """Return the profile of white- and blacklisted packages as a string."""
+    profile = """Latest configuration ({}):""".format(datetime.datetime.now())
+    for pkg, dirpath in sorted(pkg_paths.items(), key=operator.itemgetter(1)):
+        profile += "\n  * "
+        if pkg in wl_packages:
+            profile += "Whitelisted: "
+        else:
+            profile += "Blacklisted: "
+        profile += dirpath
+    profile += "\n"
+    return profile
+
+def parse_args():
+    """Parse the arguments passed to the script."""
+    parser = argparse.ArgumentParser(description="Configure a CRAM repository for a set of given packages.",
+        epilog="""The script has to be run inside the top-level directory of the CRAM repository.
+You can only pass packages which are inside the CRAM repository, otherwise the script can't know for which dependencies to configure.""")
+
+    parser.add_argument("packages", nargs='+', help="The packages to configure for.")
+    parser.add_argument("-n3p", "--no_3rdparty", action='store_true', help="This option can be supplied if for some reason not all cram_3rdparty packages should be included in the configuration.")
+    return parser.parse_args()
+
+if __name__ == '__main__':
+    args = parse_args()
+
+    crawl()
+    check_package_names()
+
+    if not check_for_high_level_packages(args.packages):
+        print(usage())
```

Comment:
This print(usage*()) gave me the **error** _python script NameError: name 'usage' is not defined_ after deleting the code line it worked flawless.